### PR TITLE
Move slot inclusion gossip validation to AttestationUtil

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description

- Essentially move some of the gossip validation to AttestationUtil to allow for fork specific logic to be replaced.

## Fixed Issue(s)
will help with #7305 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/phase0/util/AttestationUtilPhase0.java [hunk 1]
```diff
@@ -34,4 +44,70 @@ public AttestationUtilPhase0(
...
+      return Optional.of(SlotInclusionGossipValidationResult.IGNORE);
+    }
+    if (isCurrentTimeBeforeMinimumAttestationBroadcastTime(
+        attestationSlot, genesisTime, currentTimeMillis)) {
+      return Optional.of(SlotInclusionGossipValidationResult.SAVE_FOR_FUTURE);
+    }
+    return Optional.empty();
+  }
+
+  protected boolean isCurrentTimeBeforeMinimumAttestationBroadcastTime(
```
Comment (tbenr) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/phase0/util/AttestationUtilPhase0.java:70]: do we need this to be `protected`?

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/operations/AttestationData.java [hunk 2]
```diff
@@ -74,6 +75,12 @@ public UInt64 getEarliestSlotForForkChoice(final Spec spec) {
     return getSlot().plus(UInt64.ONE).max(getTarget().getEpochStartSlot(spec));
   }
 
+  public UInt64 getEarliestSlotForForkChoice(final MiscHelpers miscHelpers) {
```
Comment (tbenr) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/operations/AttestationData.java:78]: there is a bit of duplication, but at the end is ok.

**Review by tbenr — APPROVED** [review]
LGTM. just two nits

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/phase0/util/AttestationUtilPhase0.java (hunk 1 above)
Comment (StefanBratanov) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/phase0/util/AttestationUtilPhase0.java:70]: I am using the same method in the Deneb version

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/operations/AttestationData.java (hunk 2 above)
Comment (StefanBratanov) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/operations/AttestationData.java:78]: It's just that in `AttestationUtilPhase0` there is no access to `Spec`, so just using the miscHelpers instead.

**Pull request merged**